### DISCUSSION TITLE
Quarkus equivalent of Spring's @Transactional(readonly=true)?

### DISCUSSION THREAD

**Discussion by kenyee** [quarkusio/quarkus#23184] [Q&A]
https://www.baeldung.com/transaction-configuration-with-jpa-and-spring

Sorry for a bit of a Quarkus noob question, but is there something similar to
@Transactional(readonly=true)?

This page hints at it:
https://quarkus.io/guides/transaction
"I don’t need transaction when I do read only operations, it’s faster.
Wrong.
First off, just disable the transaction by marking your transaction boundary with @Transactional(NOT_SUPPORTED) (or NEVER or SUPPORTS depending on the semantic you want)."

But as I understand it, the Spring version will actually automatically switch over to a read-only data source if it's defined (e.g. a read-only replica).  The Quarkus NOT_SUPPORTED etc. variations don't seem to?

[a comment by an automated account is not shown]

**Reply by geoand** [reply]
We don't have an equivalent to `@Transactional(readonly=true)`. However I am sure @Sanne will argue that you ALWAYS want a transaction :)

**Comment by kenyee** [comment]
lol..if he wrote that section of the doc, I would expect no less :-)

From what I understand of the Spring annotation, it still is a transaction, but a read-only one...

**Comment by famod** [comment]
There have been long discussions about this, e.g. https://github.com/quarkusio/quarkus/pull/10077

This might help: https://github.com/quarkusio/quarkus/pull/10077#issuecomment-646326977

**Comment by Serkan80** [comment]
Can't you achieve the same (in Panache or on the entityManager) by setting the QueryHint.READ_ONLY to true ?
From what I've read is that doing this will disable dirty checking and all the other magic stuff that happens.

**Reply by kenyee** [reply]
yep, similar.  Would have to create two different connections and then modify one to have that QueryHint.
